src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -116,94 +116,82 @@ open class Singleton<T : Entity<T>> constructor(val entityCtor: () -> T) : Handl
         entity = entityCtor.invoke().decodeEntity(added)
     }
 
-    fun get(): T? {
-        return entity
-    }
+    fun get() = entity
 
     fun set(entity: T) {
         this.entity = entity
         val encoded = entity.encodeEntity()
         singletonSet(
-            this.particle!!.toWasmAddress(),
+            this.particle.toWasmAddress(),
             this.toWasmAddress(),
             encoded.toWasmString()
         )
     }
 
     fun clear() {
         entity = entityCtor()
-        singletonClear(this.particle!!.toWasmAddress(), this.toWasmAddress())
+        singletonClear(this.particle.toWasmAddress(), this.toWasmAddress())
     }
 }
 
 class Collection<T : Entity<T>> constructor(private val entityCtor: () -> T) : Handle(),
```

Comment:
`constructor` is not needed here, noticing it in other classes in this file as well...  It's only really needed if you're changing the visibility of the constructor.

Reply:
Done!